src/listener.ts

```diff
@@ -269,12 +274,14 @@ export class TCPListener extends EventEmitter<ListenerEvents> implements Listene
 
     const peerId = ma.getPeerId()
     const listeningAddr = peerId == null ? ma.decapsulateCode(CODE_P2P) : ma
+    const netConfig = multiaddrToNetConfig(listeningAddr)
+    const { backlog } = this.context
 
     this.status = {
       started: true,
       listeningAddr,
       peerId,
-      netConfig: multiaddrToNetConfig(listeningAddr)
+      netConfig: backlog ? { ...netConfig, backlog } : netConfig
```

Comment:
Can this be encapsulated in the `multiaddrToNetConfig` function instead?

```suggestion
      netConfig: multiaddrToNetConfig(listeningAddr, {
        backlog
      })
```

Then update it to something like:

```js
export function multiaddrToNetConfig (addr: Multiaddr, config: NetConfig = {}): NetConfig {
  const listenPath = addr.getPath()

  // unix socket listening
  if (listenPath != null) {
    if (os.platform() === 'win32') {
      // Use named pipes on Windows systems.
      return { ...config, path: path.join('\\\\.\\pipe\\', listenPath) }
    } else {
      return { ...config, path: listenPath }
    }
  }

  return {
    ...config,
    ...addr.toOptions()
  }
}
```